### PR TITLE
Fix #1468: Port error page from Gatsby to Nextjs

### LINKED ISSUE #1468: Port Error page from GatsbyJS to Next.js
**What would you like to be added**:
This is an issue to port Error page from Gatsby to NextJS

**Why would you like this to be added**:
Part of #1316, transitioning of the Telescope front-end from Gatsby to Next

### PR DESCRIPTION
**Pull request by hyunjiLeeTech**
Complete to display statusCode with error message

## Issue This PR Addresses

Fixed #1468

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

Port Error page from Gatsby to Nextjs

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by tonyvugithub** [comment]
@hyunjiLeeTech : Hi there, we are in the process of reviewing these migrated to Next PRs. Just wonder if you are still interested to work on this as we would like you to make some changes to your PR. Please let us know in case you decided to drop this then we can assign the issue to others. Thank you.

File: src/frontend/next/src/pages/404.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+const MissingPage: FC = () => {
+  return (
+    <div>
+      <h1>404</h1>
+      <p>We could not find what you were looking for</p>
+    </div>
+  );
+};
+
+export default MissingPage;
```
Comment (sonechca) [inline src/frontend/next/src/pages/404.tsx:12]: Do you need this code part? I want to know the difference from "_error.tsx". Can you add an explanation?

**Review by sonechca — COMMENTED** [review]
I checked your code by comparing it with Gatsby code. Your code was readable and nice :). I left some questions and parts that need to be modified. If you leave a comment, I'll review it again.

**Comment by sonechca** [comment]
@hyunjiLeeTech @humphd Hi, I reviewed this project but I am not sure it is correct. If @hyunjiLeeTech doesn't have time for this PR, Can I work on this?

**Comment by hyunjiLeeTech** [comment]
@sonechca Hi, please take this PR if @humphd also says it's okay.

**Comment by humphd** [comment]
@sonechca yes, you can take this.  We're going to move this to 1.6.

**Comment by birtony** [comment]
@rjayroso are you on track to get this on before our `1.6.0` release tomorrow? Any help needed?

**Comment by rjayroso** [comment]
Hi @birtony, I've been trying to tackle this for a little bit and I think I do need help - I'll bring this up tomorrow during the meeting. This current PR for the NextJS error page port is very barebones compared to it's Gatsby counterpart so I decided to try my hand at a fresh port of the error page using more of the original source code. 

Basically, the errors that I had trouble ironing out come from the following two things:

a) I think this page is still waiting on a component called `import PageBase from './PageBase';` -
In the code, it is used like so: 
```js
<PageBase title={status}>
      <DynamicImage />

      <Grid
        container
        spacing={0}
        direction="column"
        alignItems="center"
        justify="center"
        className={classes.container}
      >
        <Grid item xs={8}>
          <Card className={classes.root} elevation={6}>
            <CardContent>
              <Typography variant="h1" className={classes.h1}>
                {status}
              </Typography>

              <CreateInnerErrorContent status={status} message={message} />
            </CardContent>

            <CardActions
              style={{
                display: 'flex',
                justifyContent: 'flex-end',
                transition: 'all linear 350ms',
              }}
            >
              <Fab variant="extended" href="/" className={classes.fab}>
                <ArrowBack />
                <Typography variant="body2" className={classes.buttonText}>
                  Let&apos;s Go Back
                </Typography>
              </Fab>
            </CardActions>
          </Card>
        </Grid>
      </Grid>
    </PageBase>
```
Maybe the `PageBase` component was renamed or is planned to no longer be used and I am unaware, in that case I can try removing it and seeing what happens.

b) Second issue I'm getting is from trying to define the following property (for the life of me I just don't know even after googling for solutions):
```js
  const errorMessages = new Proxy(
    {
      400: 'We did not understand the request!',
      401: 'You are not authorized to view this page!',
      403: 'Access is not allowed for the requested page!',
      404: 'We could not find what you were looking for!',
      405: 'Method is not allowed!',
    },
    {
      get: (object, property) => {
        return property in object ? object[property] : 'Something went wrong!';
      },
    }
  );
```
On the `object[property]` line it gives me the following reason:
 `Element implicitly has an 'any' type because expression of type 'string | number | symbol' can't be used to index type '{ 400: string; 401: string; 403: string; 404: string; 405: string; }'.
  No index signature with a parameter of type 'string' was found on type '{ 400: string; 401: string; 403: string; 404: string; 405: string; }'.ts(7053)`

From what I understand, it has trouble defining that object as any single basic data type (for reference [TypeScript Data Types](https://www.typescriptlang.org/docs/handbook/basic-types.html)). Solutions that I've been thinking of were to try and explicitly define it as 'any' type but I think that would defeat the purpose of using TypeScript (maybe we can define it as 'Object' type?). I don't know if I'm overthinking this so I'd like to know what you think.

**Comment by birtony** [comment]
> 
> 
> Hi @birtony, I've been trying to tackle this for a little bit and I think I do need help - I'll bring this up tomorrow during the meeting. This current PR for the NextJS error page port is very barebones compared to it's Gatsby counterpart so I decided to try my hand at a fresh port of the error page using more of the original source code.
> 
> Basically, the errors that I had trouble ironing out come from the following two things:
> 
> a) I think this page is still waiting on a component called `import PageBase from './PageBase';` -
> In the code, it is used like so:
> 
> ```js
> <PageBase title={status}>
>       <DynamicImage />
> 
>       <Grid
>         container
>         spacing={0}
>         direction="column"
>         alignItems="center"
>         justify="center"
>         className={classes.container}
>       >
>         <Grid item xs={8}>
>           <Card className={classes.root} elevation={6}>
>             <CardContent>
>               <Typography variant="h1" className={classes.h1}>
>                 {status}
>               </Typography>
> 
>               <CreateInnerErrorContent status={status} message={message} />
>             </CardContent>
> 
>             <CardActions
>               style={{
>                 display: 'flex',
>                 justifyContent: 'flex-end',
>                 transition: 'all linear 350ms',
>               }}
>             >
>               <Fab variant="extended" href="/" className={classes.fab}>
>                 <ArrowBack />
>                 <Typography variant="body2" className={classes.buttonText}>
>                   Let&apos;s Go Back
>                 </Typography>
>               </Fab>
>             </CardActions>
>           </Card>
>         </Grid>
>       </Grid>
>     </PageBase>
> ```
> 
> Maybe the `PageBase` component was renamed or is planned to no longer be used and I am unaware, in that case I can try removing it and seeing what happens.
> 
> b) Second issue I'm getting is from trying to define the following property (for the life of me I just don't know even after googling for solutions):
> 
> ```js
>   const errorMessages = new Proxy(
>     {
>       400: 'We did not understand the request!',
>       401: 'You are not authorized to view this page!',
>       403: 'Access is not allowed for the requested page!',
>       404: 'We could not find what you were looking for!',
>       405: 'Method is not allowed!',
>     },
>     {
>       get: (object, property) => {
>         return property in object ? object[property] : 'Something went wrong!';
>       },
>     }
>   );
> ```
> 
> On the `object[property]` line it gives me the following reason:
> `Element implicitly has an 'any' type because expression of type 'string | number | symbol' can't be used to index type '{ 400: string; 401: string; 403: string; 404: string; 405: string; }'. No index signature with a parameter of type 'string' was found on type '{ 400: string; 401: string; 403: string; 404: string; 405: string; }'.ts(7053)`
> 
> From what I understand, it has trouble defining that object as any single basic data type (for reference [TypeScript Data Types](https://www.typescriptlang.org/docs/handbook/basic-types.html)). Solutions that I've been thinking of were to try and explicitly define it as 'any' type but I think that would defeat the purpose of using TypeScript (maybe we can define it as 'Object' type?). I don't know if I'm overthinking this so I'd like to know what you think.

Hey @rjayroso! As discussed during today's triage, let's set up a group coding session sometime soon to figure this out. In the meanwhile, feel free to create a new PR and push any code you have.

**Comment by rjayroso** [comment]
This PR is continued in #1746

**Pull request closed without merging**